src/security/Guardian.sol

```diff
@@ -0,0 +1,179 @@
+                            STORAGE
+    //////////////////////////////////////////////////////////////
+    */
+    bool public repayPaused;
+    bool public withdrawPaused;
+    bool public borrowPaused;
+    bool public depositPaused;
+    bool public liquidationPaused;
+    uint256 public pauseTimestamp;
+
+    constructor() Owned(msg.sender) {
+        repayPaused = false;
+        withdrawPaused = false;
+        borrowPaused = false;
+        depositPaused = false;
+    }
+
+    /*
+    //////////////////////////////////////////////////////////////
+                            MODIFIERS
+    //////////////////////////////////////////////////////////////
+    */
+
+    /**
+     * @dev Throws if called by any account other than the owner.
+     */
+    modifier onlyGuardian() {
+        require(msg.sender == guardian, "Guardian: Only guardian can call this function");
+        _;
+    }
+
+    /**
+     * @dev Throws if repay is paused.
+     */
+    modifier whenRepayNotPaused() {
+        require(!repayPaused, "Guardian: repay paused");
+        _;
+    }
+
+    /**
+     * @dev Throws if withdraw is paused.
+     */
+    modifier whenWithdrawNotPaused() {
+        require(!withdrawPaused, "Guardian: withdraw paused");
+        _;
+    }
+
+    /**
+     * @dev Throws if borrow is paused.
+     */
+    modifier whenBorrowNotPaused() {
+        require(!borrowPaused, "Guardian: borrow paused");
+        _;
+    }
+
+    /**
+     * @dev Throws if deposit is paused.
+     */
+    modifier whenDepositNotPaused() {
+        require(!depositPaused, "Guardian: deposit paused");
+        _;
+    }
+
+    /**
+     * @dev Throws if liquidation is paused.
+     */
+    modifier whenLiquidationNotPaused() {
+        require(!liquidationPaused, "Guardian: liquidation paused");
+        _;
+    }
+
+    /**
+     * @param guardian_ The address of the new guardian.
+     * @dev Allows onlyOwner to change the guardian address.
+     */
+    function changeGuardian(address guardian_) external onlyOwner {
+        guardian = guardian_;
+        emit GuardianChanged(guardian, guardian_);
+    }
+
+    /**
+     * @dev This function can be called by the guardian to pause all functionality in the event of an emergency.
+     *      This function pauses repay, withdraw, borrow, deposit and liquidation.
+     *      This function can only be called by the guardian.
+     *      Guardian can only pause the protocol once 32 days past from the last pause. This is to prevent
+     *  guardian from pausing the protocol too often. And giving unpause time for users to trigger.
+     *  When protocol is paused for 30 days only owner role has a right to unpause the protocol. After 30 days,
+     *  any user can unpause the protocol. This flow gives at least 2 days to unpause the protocol for any user
+     *  since guardian can only trigger pause once every 32 days after the previous pause event.
+     */
+    function pause() external onlyGuardian {
```

Comment:
Should the owner also be able to pause the protocol (edge case where guardian multisig has different signers as owner multisig)

Reply:
I think no, i do not see a reason to give pause power to 2 addresses.